gapi/gapi.go

```
@@ -0,0 +1,185 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+)
+
+// DAO acts as the gapi DAO
+type DAO interface {
+	GenerateFireDoc(title string) (string, error)
+}
+
+// DAOImpl defines the properties of the DAO
+type DAOImpl struct {
+	Email      string
+	PrivateKey []byte
+	Client     *http.Client
+	FolderID   string
+}
+
+// NewDAO returns a DAO including a Google API authenticated HTTP client
+func NewDAO(email string, privateKey string, folderID string) DAO {
+	scopes := []string{"https://www.googleapis.com/auth/drive", "https://www.googleapis.com/auth/documents"}
+	conf := &jwt.Config{
+		Email:      email,
+		PrivateKey: []byte(privateKey),
+		Scopes:     scopes,
+		TokenURL:   google.JWTTokenURL,
+	}
+	client := conf.Client(oauth2.NoContext)
+	return &DAOImpl{
+		Email:      conf.Email,
+		PrivateKey: conf.PrivateKey,
+		Client:     client,
+		FolderID:   folderID,
+	}
+}
+
+func jsonDecode(body io.ReadCloser) map[string]interface{} {
+	data := make(map[string]interface{})
+	json.NewDecoder(body).Decode(&data)
```

are you swallowing an error here?

---

Good catch, ty.